libs/designer-ui/src/lib/floatingactionmenu/index.tsx

```diff
@@ -132,9 +156,42 @@ export const FloatingActionMenu = (props: FloatingActionMenuProps): JSX.Element
     }
   };
 
-  const handleMenuItemSelected = (_item: FloatingActionMenuItem) => {
+  const enterVariableNamePlaceholder = intl.formatMessage({
+    defaultMessage: 'Enter variable name',
+    description: 'Placeholder for variable name for new dynamically added parameter',
+  });
+
+  const addNewDynamicallyAddedParameter = (item: FloatingActionMenuItem) => {
+    const { icon, type, placeholder } = item;
+    const format =
+      type === DynamicallyAddedParameterType.Date || type === DynamicallyAddedParameterType.Email ? type.toLowerCase() : undefined;
+    dynamicParameterProps.push({
+      icon,
+      schemaKey: guid(), // TODO: generate schemaKey
+      properties: {
+        description: placeholder,
+        format,
+        title: enterVariableNamePlaceholder, // TODO: generate title based on schemaKey
+        type,
+        'x-ms-content-hint': type,
+        'x-ms-dynamically-added': true,
+      },
+      required: true, // TODO: add functionality to allow making parameters optional
```

Comment:
In v1 designer, this option looks like this in the menu -- 

![image](https://user-images.githubusercontent.com/10837129/233415343-7b6f9d08-f6ce-41f8-9f6c-8f1badd05ec0.png)

Reply:
Not blocking right now.